dask_match/tests/test_collection.py

```diff
@@ -139,12 +139,85 @@ def test_repr(ddf):
 
 
 def test_columns_traverse_filters(df, ddf):
-    result = optimize(ddf[ddf.x > 5].y)
+    result = optimize(ddf[ddf.x > 5].y, fuse=False)
     expected = ddf.y[ddf.x > 5]
 
     assert str(result) == str(expected)
 
 
+def test_optimize_fusion(ddf):
+    ddf2 = (ddf["x"] + ddf["y"]) - 1
+    unfused = optimize(ddf2, fuse=False)
+    fused = optimize(ddf2, fuse=True)
+
+    # Should only get one task per partition
+    assert len(fused.dask) == ddf.npartitions
+    assert_eq(fused, unfused)
+
+    # Check that we still get fusion when
+    # non-blockwise operations are added
+    ddf3 = ddf2.sum()
+    unfused = optimize(ddf3, fuse=False)
+    fused = optimize(ddf3, fuse=True)
+
+    assert len(fused.dask) < len(unfused.dask)
+    assert_eq(fused, unfused)
+
+    # Check that we still get fusion
+    # after a non-blockwise operation as well
+    fused_2 = optimize((ddf3 + 10) - 5, fuse=True)
+    # The "+10 and -5" ops should get fused
+    assert len(fused_2.dask) == len(fused.dask) + 1
+
+
+@pytest.mark.parametrize("persist", [True, False])
+def test_optimize_fusion_many(ddf, persist):
```

Comment:
These tests seem to take far longer than others. While we're this early in the project I'd love to keep tests snappy. I think that that will keep development fast.  Do you have thoughts on why these are slow?